contracts/Gate.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+pragma solidity 0.7.1;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+
+import "./interfaces/IGate.sol";
+import "./interfaces/IERC1155.sol";
+
+/**
+ * @title Gate contract between Boson router and ERC1155NonTransferable
+ * Enables conditional commit
+ */
+
+contract Gate is IGate, Ownable, Pausable {
+    mapping(uint256 => uint256) private voucherToToken;
+    mapping(address => mapping(uint256 => bool)) private isDeactivated; // mapping user => voucherSet => bool
+
+    IERC1155 private nonTransferableTokenContract;
+    address private bosonRouterAddress;
+  
+    modifier onlyFromRouter() {
+        require(bosonRouterAddress != address(0), "UNSPECIFIED_BR"); //hex"20" FISSION.code(FISSION.Category.Find, FISSION.Status.NotFound_Unequal_OutOfRange)
+        require(msg.sender == bosonRouterAddress, "UNAUTHORIZED_BR"); //hex"10" FISSION.code(FISSION.Category.Permission, FISSION.Status.Disallowed_Stop)
+        _;
+    }
+
+    /**
+     * @notice For a given _tokenIdSupply, it tells on which NFT it depends
+     * @param _tokenIdSupply an ID of a supply token (ERC-1155) [voucherSetID]
+     * @return quest NFT token ID
+     */
+    function getNftTokenId(uint256 _tokenIdSupply) external view override returns (uint256) {
+        return voucherToToken[_tokenIdSupply];
+    }
+
+    /**
+     * @notice Sets the contract, where gate contract checks if quest NFT token exists
+     * @param _nonTransferableTokenContractAddress address of a non-transferable token contract
+     */
+    function setNonTransferableTokenContract(
+        address _nonTransferableTokenContractAddress
+    ) external override onlyOwner {
+        nonTransferableTokenContract = IERC1155(
+            _nonTransferableTokenContractAddress
+        );
+
+        emit LogNonTransferableContractSet(_nonTransferableTokenContractAddress);
+    }
+
+    /**
```

Comment:
I think we should probably throw in a getter, it is a lot nicer than having to dig around for a log ..

Reply:
Was suggested also by Ludo and already implemented.